go/vt/vtgate/engine/route.go

```diff
@@ -511,6 +517,15 @@ func (route *Route) execUpdateEqualChangedVindex(vcursor VCursor, query string,
 	return result, nil
 }
 
+func (route *Route) execDeleteSharded(vcursor VCursor, bindVars map[string]*querypb.BindVariable) (*sqltypes.Result, error) {
+	params, err := route.paramsAllShards(vcursor, bindVars)
+	if err != nil {
+		return nil, err
+	}
+	shardQueries := route.getShardQueries(route.Query, params)
```

Comment:
Need to call `sqlannotation.AnnotateIfDML` here. We also need executor_dml_test cases to verify that the annotations were added during engine execution.

The feature will still work without the annotations (due to failsafes), but we still have to follow the rules.